plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -191,9 +202,9 @@ public void doMigrate(ConnectorSession session, String schemaName, String tableN
             throw new TrinoException(NOT_SUPPORTED, "The table is already an Iceberg table");
         }
 
-        Schema schema = toIcebergSchema(concat(hiveTable.getDataColumns().stream(), hiveTable.getPartitionColumns().stream()).toList());
-        NameMapping nameMapping = MappingUtil.create(schema);
         HiveStorageFormat storageFormat = extractHiveStorageFormat(hiveTable.getStorage().getStorageFormat());
+        Schema schema = toIcebergSchema(concat(hiveTable.getDataColumns().stream(), hiveTable.getPartitionColumns().stream()).toList(), storageFormat);
```

Comment:
`storageFormat` -> `toIcebergFileFormat(storageFormat)`
This way we can pass in `IcebergFileFormat` instead of `HiveStorageFormat` to `toIcebergSchema`. HiveStorageFormat has more formats than avro/parquet/orc which we don't handle explicitly in `toIcebergSchema`.